File: src/test/java/quebec/virtualite/backend/services/rest/WheelDTOTest.java
package quebec.virtualite.backend.services.rest;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

import static quebec.virtualite.backend.TestConstants.BAD_WHEEL_DTO;
import static quebec.virtualite.backend.TestConstants.WHEEL_DTO;
import static quebec.virtualite.backend.services.utils.TestUtils.assertInvalid;
import static quebec.virtualite.backend.services.utils.TestUtils.assertValid;

@ExtendWith(MockitoExtension.class)
class WheelDTOTest
{
    @Test
    void validate()
    {
        assertValid(WHEEL_DTO);

        assertInvalid(WHEEL_DTO.withBrand(null));
        assertInvalid(WHEEL_DTO.withBrand(""));

        assertInvalid(WHEEL_DTO.withName(null));
        assertInvalid(WHEEL_DTO.withName(""));

        assertInvalid(BAD_WHEEL_DTO);
    }
}

File: src/test/java/quebec/virtualite/backend/services/rest/WheelDTOTest.kt
package quebec.virtualite.backend.services.rest

import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner
import quebec.virtualite.backend.TestConstants.BAD_WHEEL_DTO
import quebec.virtualite.backend.TestConstants.WHEEL_DTO
import quebec.virtualite.backend.services.utils.TestUtils.assertInvalid
import quebec.virtualite.backend.services.utils.TestUtils.assertValid

@RunWith(MockitoJUnitRunner::class)
class WheelDTOTest
{
    @Test
    fun validate()
    {
        assertValid(WHEEL_DTO)

        assertInvalid(WHEEL_DTO.copy(brand = null))
        assertInvalid(WHEEL_DTO.copy(brand = ""))

        assertInvalid(WHEEL_DTO.copy(name = null))
        assertInvalid(WHEEL_DTO.copy(name = ""))

        assertInvalid(BAD_WHEEL_DTO)
    }
}
